File: iOS/Source/LiveRoom/Data/MusicInfo.swift
//
//  File.swift
//  TUILiveKit
//
//  Created by WesleyLei on 2023/11/20.
//

import Foundation

class MusicInfo {
    var id: Int32
    var name: String
    var path: String
    let isPlaying: Observable<Bool> = Observable(false)
    var pitch: Float = 0.0

    init() {
        self.id = -1
        self.name = ""
        self.path = ""
    }
    
    init(id: Int32, name: String, path: String) {
        self.id = id
        self.name = name
        self.path = path
    }

    static func defaultMusicInfoList() -> [MusicInfo] {
        return [MusicInfo(id: 1,
                          name: .localized("live.music.cheerful"),
                          path: "https://dldir1.qq.com/hudongzhibo/TUIKit/resource/music/PositiveHappyAdvertising.mp3"),
                MusicInfo(id: 2,
                          name: .localized("live.music.melancholy"),
                          path: "https://dldir1.qq.com/hudongzhibo/TUIKit/resource/music/SadCinematicPiano.mp3"),
                MusicInfo(id: 3,
                          name: .localized("live.music.wonder.world"),
                          path: "https://dldir1.qq.com/hudongzhibo/TUIKit/resource/music/WonderWorld.mp3"), ]
    }
}

File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/liveroom/data/MusicInfo.java
package com.trtc.uikit.livekit.liveroom.data;

import com.trtc.tuikit.common.livedata.LiveData;

public class MusicInfo {
    public static final int INVALID_ID = -1;

    public int               id        = INVALID_ID;
    public String            name;
    public String            path;
    public LiveData<Boolean> isPlaying = new LiveData<>(false);
    public LiveData<Float>   pitch     = new LiveData<>(0.0f);

    public MusicInfo() {
    }

    public MusicInfo(int id, String name, String musicPath) {
        this.id = id;
        this.name = name;
        this.path = musicPath;
    }
}
